pdns/dnsdistdist/dnsdist-mac-address.hh

```diff
@@ -0,0 +1,48 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#pragma once
+
+#include "circular_buffer.hh"
+#include "iputils.hh"
+#include "lock.hh"
+
+namespace dnsdist
+{
+class MacAddressesCache
+{
+public:
+  static int get(const ComboAddress& ca, char* dest, size_t len);
+
+private:
+  struct Entry
+  {
+    ComboAddress ca;
+    char mac[6];
```

Comment:
Move this field down, so that alignment of the time_t field is better.
Also: why char[6] and not array<uint8_t, 6>? clang-tidy does not like "C stye" arrays at all...

Reply:
That's a very good point, I'll look into it!

Reply:
Done: